src/main/java/de/scimeda/keycloak/events/logging/LastLoginEventListenerProvider.java

```diff
@@ -35,6 +36,12 @@ public void onEvent(Event event) {
             if (user != null) {
                 log.info("Updating last login status for user: " + event.getUserId());
 
+                Map userAttrs = user.getAttributes();
```

Comment:
This seems to cause the type mismapping.
 ```suggestion
                Map<String, List<String>> userAttrs = user.getAttributes();
```
Afterwards this should return a proper List object from which you can get the first value.

Not sure whether it is necessary to sanity check there is actually anything in this list.
The event listener shouldn't let this happen but a user or admin may do so (unless you set them as read only attributes but I haven't tested this feature yet).

Reply:
Thanks for your help and pointing me to the obvious. I've fixed the List type.

Should I add the sanity check or do you want to check the user attributes API first?